fix: squash splash calculation bug

br-Chan opened the pull request:
I've simply removed the rounding of splash attack result - I'm now testing to make sure this is all that's required. 

[Reddit post](https://www.reddit.com/r/Polytopia/comments/1m0k11v/is_this_damage_calculator_up_to_date_i_did_this/):
<img width="1005" height="857" alt="image" src="https://github.com/user-attachments/assets/3e155c45-20b9-40a5-b379-dc828b08ab98" />

br-Chan commented:
@amigobrewbrew Hi, here's a fix! Let me know if you find any errors - I've done manual testing in the above comment.

Current calculations show the decimal 0.5. We can instead round down the splash attack result so that it should 1 hp like it does in-game and remove any confusion. The polycalc Discord bot keeps the 0.5 display without rounding, however. I'm happy with either!

amigobrewbrew commented:
@br-Chan Looks great thanks!

amigobrewbrew commented:
@br-Chan Nice great stuff. Keeping it 0.5 will be better. It gives more insight in the actual HP the unit has... I also added that input field can now be 1.5 etc and there was a "XPLD did not set poison" issue. Also fixed. All deployed!

br-Chan commented:
> @br-Chan Nice great stuff. Keeping it 0.5 will be better. It gives more insight in the actual HP the unit has... I also added that input field can now be 1.5 etc and there was a "XPLD did not set poison" issue. Also fixed. All deployed!

Awesome, thanks!
